### ISSUE TITLE
execute_command() behaves strangely in cluster mode

### ISSUE THREAD

**Issue by suxb201** [redis/redis-py#2095]
**Version**: redis==4.2.2
**Platform**: MacOS with python==3.10
**Description**: `execute_command` behaves strangely in cluster mode.

run no-key command:
```python
# works fine
rc.cluster_info()
rc.execute_command("CLUSTER INFO")

# RedisClusterException: No way to dispatch this command to Redis Cluster. Missing key.
rc.execute_command("CLUSTER", "INFO")
rc.execute_command("cluster info")
rc.execute_command("cluster", "info")
```

**Comment by chayim** [comment]
@suxb201 can you try using the various functions intended for this purpose, on the cluster interface? I'm asking because underneath execute_command should be used for commands that are unimplemented, and there are function differences associated with the connection.

**Comment by chayim** [comment]
Hi @suxb201 have you tried using the commands directly, for the cluster interface rather than execute command? While perhaps there's improvement for raw commands we can do here (clearly!), at least this should get you over your hump.

**Comment by suxb201** [comment]
We have some scenarios where we use string concatenation commands, so execute_command() is the best choice.

**Comment by suxb201** [comment]
Thanks to @chayim and @dvora-h , the problem has been solved in #2097.
